### DISCUSSION TITLE
What is the correct usage of OP conductor?

### DISCUSSION THREAD

**Discussion by Padraic-O-Mhuiris** [ethereum-optimism/developers#525] [Node Operators] [answered]
### Are you running the most up to date node software?

- [X] Yes

### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

Some of my questions are similar to #313 which has remained unanswered for the most part. Trying to understand how to successfully use OP conductor with these specific questions?

1) My initial impression is that the node runs like a sidecar similar to `op-batcher` and `op-proposer`. Is that the case? So a `1:1` pairing with each `op-node`
2) If multiple `op-node` instances are ran, should the `--sequencer.enabled` flag be set to true. I understand this may be redundant if the conductor enables/disables over rpc?
3) If there are "n" `op-node` instances, is it the case that there must be "n" instances of `op-batcher` and `op-proposer`
4) It's not clear how `op-conductor` instances find each other? I see the `--consensus.addr` flag which I understand must be a public IP like `--p2p.advertise.ip` in `op-node` but I can't seem to find in the code how other `op-conductor` instances are found.

Would be very grateful for a reply.

### Protocol Description

N/A

**Comment by joohhnnn** [comment, marked as answer]
1. Yes, at an abstract level, but for network robustness, a sequencer typically has mechanisms like op-conductor to ensure stable operation, which may involve multiple internal instances.

2. Triggered via RPC — the --sequencer.enabled still needs to be set. You can see this at: [GitHub Link](https://github.com/ethereum-optimism/optimism/blob/acafdb9eed71d07dd9dbf3944195840cbd61fd8b/op-node/rollup/driver/state.go#L586)

3. I'm not sure, but currently, op-conductor is only for sequencer part. Theoretically, op-batcher and op-proposer could also have multiple instances, but personally, I don’t see much necessity since the batcher and proposer mainly interact with L1 and do not have the same immediacy requirements as a sequencer.

4. It's mainly implemented in the raft library. Op-conductor merely provides configurations like ServerID and address, which you can understand as: if the configurations are the same, they will join the same raft consensus network. For example: https://github.com/ethereum-optimism/optimism/blob/acafdb9eed71d07dd9dbf3944195840cbd61fd8b/op-conductor/consensus/raft.go#L80

**Reply by Padraic-O-Mhuiris** [reply]
This is fantastic information, many thanks
